core/src/main/resources/error/error-classes.json

```diff
@@ -196,6 +196,9 @@
   "UNSUPPORTED_OPERATION" : {
     "message" : [ "The operation is not supported: <operation>" ]
   },
+  "UNSUPPORTED_SAVE_MODE" : {
+    "message" : [ "The saveMode is not supported: <saveMode> (<pathExists>)" ]
```

Comment:
The message is not clear from user side, I believe. Let's introduce two sub-classes EXISTENT_PATH and NON_EXISTENT_PATH. Messages should be something like:
1. The save mode <saveMode> is not supported for: an existent path.
2. The save mode <saveMode> is not supported for: a not existent path.

What's does true or false in the error message mean is not clear from my point of view.

See https://github.com/apache/spark/pull/36307 for an example.